### PR TITLE
Stringify key :message in the base formatter

### PR DESCRIPTION
**Pull request by jcmfernandes**
Doing the following (an example taken from the README
file)

logger.debug message: 'test', foo: 'bar'

will fail for any logger whose the formatter's class
inherits from LogStashLogger::Formatter::Base when
LogStashLogger.configuration.max_message_size is
set. This happens because event['message'.freeze].byteslice
(previously on line 49) will fail as
event['message'.freeze] will return nil.

### CODE REVIEW

**Comment by dwbutler** [comment]
Nice catch. And good call on treating the input as immutable as well.

I dove into the source code of `LogStash::Event` and discovered that most of the "built in" fields are expected to use string keys. So it might make sense to convert all the keys to strings.

Example of the problem:

```ruby
LogStash::Event.new(tags: ['foo']).tags
# => nil
pry(main)> LogStash::Event.new('tags' => ['foo']).tags
# => ["foo"]
```

**Comment by jcmfernandes** [comment]
Thanks! 😺 

I agree. Should we deep stringify hashes or just root keys?

File: lib/logstash-logger/formatter/base.rb [hunk 1]
```diff
@@ -25,7 +25,9 @@ def build_event(message, severity, time)
                   when LogStash::Event
                     data.clone
                   when Hash
-                    event_data = data.merge("@timestamp".freeze => time)
+                    event_data = data.clone
+                    event_data['message'.freeze] = event_data.delete(:message) if event_data.key?(:message)
```
Comment (stereobooster) [inline lib/logstash-logger/formatter/base.rb]: Constanize strings as well e.g. 
```ruby
MESSAGE_FIELD = 'message'.freeze`
event_data[MESSAGE_FIELD]
```

**Comment by dwbutler** [comment]
There is a very specific set of keys that `LogStash::Event` expects to be strings. There wouldn't be any value in stringifying all keys, or deep converting keys.

Since this fixes an immediate bug, I'll merge it as-is and implement a more generic solution later.

File: lib/logstash-logger/formatter/base.rb (hunk 1 above)
Comment (dwbutler) [inline lib/logstash-logger/formatter/base.rb]: My recollection is that Ruby will optimize this sort of code anyway, and therefore extracting to a constant isn't necessary. Can't find a source for this.

Comment (stereobooster) [inline lib/logstash-logger/formatter/base.rb]: Yep. I missed this optimisation somehow. Here is the proof (at least for `ruby 2.2.6p396 (2016-11-15 revision 56800)`):
```ruby
irb(main):001:0> "asd".object_id
=> 70178538223500
irb(main):002:0> "asd".object_id
=> 70178538189060
irb(main):003:0> "asd".freeze.object_id
=> 70178538223620
irb(main):004:0> "asd".freeze.object_id
=> 70178538223620
```

Comment (stereobooster) [inline lib/logstash-logger/formatter/base.rb]: but for ruby 2.0.0p247 (2013-06-27 revision 41674)

```ruby
irb(main):001:0> "asd".object_id
=> 70313257930780
irb(main):002:0> "asd".object_id
=> 70313241344880
irb(main):003:0> "asd".freeze.object_id
=> 70313241338520
irb(main):004:0> "asd".freeze.object_id
=> 70313241348500
irb(main):005:0> "asd".freeze.object_id
```

And taking into account that library claims to support ruby 2.0 constanization still makes sense

Comment (dwbutler) [inline lib/logstash-logger/formatter/base.rb]: Thanks for the legwork. And good point, LogStashLogger support goes back to 1.9, so it should definitely do whatever optimizations are necessary.

**Comment by jcmfernandes** [comment]
Thank you!

**Comment by dwbutler** [comment]
Fix released in 0.24.1. Thanks!

**Pull request merged**